src/tools/rbd/action/ImageCache.cc

```diff
@@ -0,0 +1,80 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "tools/rbd/ArgumentTypes.h"
+#include "tools/rbd/Shell.h"
+#include "tools/rbd/Utils.h"
+#include "include/types.h"
+#include "include/rbd_types.h"
+#include "include/stringify.h"
+#include "common/errno.h"
+#include "common/Formatter.h"
+#include <iostream>
+#include <boost/program_options.hpp>
+
+namespace rbd {
+namespace action {
+namespace image_cache {
+
+namespace at = argument_types;
+namespace po = boost::program_options;
+
+void get_arguments(po::options_description *positional,
+                   po::options_description *options) {
+  at::add_image_spec_options(positional, options,
+				at::ARGUMENT_MODIFIER_NONE);
+  at::add_image_id_option(options);
+  /* We'd need encoders for each image cache spec to do this */
+  at::add_format_options(options);
+}
+
+int execute_discard(const po::variables_map &vm,
+                    const std::vector<std::string> &ceph_global_init_args) {
+  size_t arg_index = 0;
+  std::string pool_name;
+  std::string namespace_name;
+  std::string image_name;
+  std::string snap_name;
```

Comment:
Nit: no need to support snapshots, right?

Reply:
Yes, it is limited in utils::get_pool_image_snapshot_names(), utils::SNAPSHOT_PRESENCE_NONE is given to disallow snapshot.

Reply:
You can pass a `nullptr` in that case.

Reply:
OK